calculator.js

```diff
@@ -265,18 +278,246 @@ input_element.addEventListener("click", event =>{
 //CALCULATOR
 function calculator(button){
     if(button.type == "operator"){
+        data.operation.push( button.symbol );
+        data.formula.push( button.formula );
 
     }else if(button.type == "number"){
+        data.operation.push( button.symbol );
+        data.formula.push( button.formula );
 
     }else if(button.type == "trigo_function"){
+        data.operation.push(button.symbol + "(");
+        data.formula.push(button.formula);
 
     }else if(button.type == "math_function"){
+        let symbol, formula;
+
+        if (button.name == "factorial"){
+            symbol = "!";
+            formula = button.formula;
+
+            data.operation.push(symbol);
+            data.formula.push(formula);
+        } else if (button.name == "power"){
```

Comment:
Same here: else if (button.name == "power" || button.name == "square")

Reply:
not really, the "square" button closes the formula with: 2)... in other word if you press 5 and then the "square" botton, the calc will write: 5^(2)... køntash!?